src/FlatDirectoryCertificateStore.cs

```diff
@@ -0,0 +1,233 @@
+    }
+
+    /// <inheritdoc/>
+    public Task<X509CRLCollection> EnumerateCRLs()
+    {
+        return _innerStore.EnumerateCRLs();
+    }
+
+    /// <inheritdoc/>
+    public Task<X509CRLCollection> EnumerateCRLs(X509Certificate2 issuer, bool validateUpdateTime = true)
+    {
+        return _innerStore.EnumerateCRLs(issuer, validateUpdateTime);
+    }
+
+    /// <inheritdoc/>
+    public async Task<X509Certificate2Collection> FindByThumbprint(string thumbprint)
+    {
+        var certificatesCollection = await _innerStore.FindByThumbprint(thumbprint).ConfigureAwait(false);
+
+        if (!_innerStore.Directory.Exists)
+        {
+            return certificatesCollection;
+        }
+
+        foreach (FileInfo file in _innerStore.Directory.GetFiles('*' + CrtExtension))
+        {
+            try
+            {
+                X509Certificate2Collection certificates = new X509Certificate2Collection();
+                certificates.ImportFromPemFile(file.FullName);
+                foreach (X509Certificate2 certificate in certificates)
+                {
+                    if (string.Equals(certificate.Thumbprint, thumbprint, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Utils.LogInfo("Find by thumbprint: {0} - found", thumbprint);
+                        certificatesCollection.Add(certificate);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Utils.LogError(e, "Could not load certificate from file: {0}", file.FullName);
+            }
+        }
+
+        return certificatesCollection;
+    }
+
+    /// <inheritdoc/>
+    public Task<StatusCode> IsRevoked(X509Certificate2 issuer, X509Certificate2 certificate)
+    {
+        return _innerStore.IsRevoked(issuer, certificate);
+    }
+
+    /// <inheritdoc/>
+    public async Task<X509Certificate2> LoadPrivateKey(string thumbprint, string subjectName, string password)
+    {
+        if (!_innerStore.Directory.Exists)
+        {
+            return await _innerStore.LoadPrivateKey(thumbprint, subjectName, password).ConfigureAwait(false);
+        }
+
+        foreach (FileInfo file in _innerStore.Directory.GetFiles('*' + CrtExtension))
+        {
+            try
+            {
+                FileInfo keyFile = new FileInfo(file.FullName.Replace(CrtExtension, KeyExtension, StringComparison.OrdinalIgnoreCase));
+                if (keyFile.Exists)
+                {
+                    using X509Certificate2 certificate = new X509Certificate2(file.FullName);
+                    if (!MatchCertificate(certificate, thumbprint, subjectName))
+                    {
+                        continue;
+                    }
+
+                    X509Certificate2 privateKeyCertificate = X509Certificate2.CreateFromPemFile(file.FullName, keyFile.FullName);
+
+                    Utils.LogInfo("Loading private key succeeded for {0} - {1}", thumbprint, subjectName);
```

Comment:
>{0} - {1} [](http://example.com/codeflow?start=69&length=9)

If possible, could you pls change the numeric placeholders in this file to use strings instead? they get flagged by the analyzers and we need to migrate the remaining ones someday anyway.

Reply:
done

Reply:
done